From 527d47a77282ef38de5e2f131470ada5374c0348 Mon Sep 17 00:00:00 2001
From: Casey Schaufler <casey@schaufler-ca.com>
Date: Thu, 7 Nov 2013 16:23:09 -0800
Subject: [PATCH 06/24] Smack sharing of /run

Make /run a transmuting directory to enable systemd
communications with services in the User domain.

Add more devices that require general access.
Mount /tmp publicly accessable.

Signed-off-by: Casey Schaufler <casey.schaufler@intel.com>

Change-Id: I85f4b386978b7a993938557f8739067f3183e432
---
 rules/55-udev-smack-default.rules | 2 ++
 src/core/mount-setup.c            | 4 +++-
 units/tmp.mount                   | 2 +-
 3 files changed, 6 insertions(+), 2 deletions(-)

diff --git a/rules/55-udev-smack-default.rules b/rules/55-udev-smack-default.rules
index be8c4f8..dd45c51 100644
--- a/rules/55-udev-smack-default.rules
+++ b/rules/55-udev-smack-default.rules
@@ -6,6 +6,8 @@ KERNEL=="console", SMACK="*"
 KERNEL=="kmsg", SMACK="*"
 KERNEL=="video*", SMACK="*"
 KERNEL=="card*", SMACK="*"
+KERNEL=="ptmx", SMACK="*"
+KERNEL=="tty", SMACK="*"
 
 SUBSYSTEM=="graphics", GROUP="video", SMACK="*"
 SUBSYSTEM=="drm", GROUP="video", SMACK="*"
diff --git a/src/core/mount-setup.c b/src/core/mount-setup.c
index cc2633e..c7e4dbb 100644
--- a/src/core/mount-setup.c
+++ b/src/core/mount-setup.c
@@ -91,8 +91,10 @@ static const MountPoint mount_table[] = {
         { "devpts",     "/dev/pts",                  "devpts",     "mode=620,gid=" STRINGIFY(TTY_GID), MS_NOSUID|MS_NOEXEC,
           NULL,       MNT_IN_CONTAINER },
 #ifdef HAVE_SMACK
-        { "tmpfs",      "/run",                      "tmpfs",      "mode=755,smackfsroot=*", MS_NOSUID|MS_NODEV|MS_STRICTATIME,
+        { "tmpfs",      "/run",                      "tmpfs",      "mode=755,smackfstransmute=System::Run", MS_NOSUID|MS_NODEV|MS_STRICTATIME,
           use_smack,  MNT_FATAL },
+        { "tmpfs",      "/sys/fs/cgroup",            "tmpfs",      "mode=755,smackfsroot=*", MS_NOSUID|MS_NOEXEC|MS_NODEV|MS_STRICTATIME,
+          use_smack,  MNT_IN_CONTAINER },
 #endif
         { "tmpfs",      "/run",                      "tmpfs",      "mode=755", MS_NOSUID|MS_NODEV|MS_STRICTATIME,
           NULL,       MNT_FATAL|MNT_IN_CONTAINER },
diff --git a/units/tmp.mount b/units/tmp.mount
index 00a0d28..415c07f 100644
--- a/units/tmp.mount
+++ b/units/tmp.mount
@@ -18,4 +18,4 @@ Before=local-fs.target umount.target
 What=tmpfs
 Where=/tmp
 Type=tmpfs
-Options=mode=1777,strictatime
+Options=mode=1777,strictatime,smackfsroot=*
-- 
1.8.4.5

